climada/hazard/centroids/centr.py

```diff
@@ -79,280 +52,215 @@ class Centroids():
-            base_file = NATEARTH_CENTROIDS[res_as]
-
-        centroids = Centroids.from_hdf5(base_file)
-        if centroids.meta:
-            xres, xshear, xoff, yshear, yres, yoff = centroids.meta['transform'][:6]
-            shape = (centroids.meta['height'], centroids.meta['width'])
-            if yres > 0:
-                # make sure y-orientation is negative
-                centroids.meta['transform'] = rasterio.Affine(xres, xshear, xoff, yshear,
-                                                              -yres, yoff + (shape[0] - 1) * yres)
-                # flip y-axis in data arrays
-                for name in ["region_id", "dist_coast"]:
-                    if not hasattr(centroids, name):
-                        continue
-                    data = getattr(centroids, name)
-                    if data.size == 0:
-                        continue
-                    setattr(centroids, name, np.flipud(data.reshape(shape)).reshape(-1))
-        if land:
-            land_reg_ids = list(range(1, 1000))
-            land_reg_ids.remove(10)  # Antarctica
-            centroids = centroids.select(reg_id=land_reg_ids)
-
-        centroids.check()
-        return centroids
+        self.gdf.to_crs(DEF_CRS, inplace=True)
 
-    @classmethod
-    def from_geodataframe(cls, gdf, geometry_alias='geom'):
-        """Create Centroids instance from GeoDataFrame.
-
-        .. deprecated:: 3.3
-           This method will be removed in a future version. Pass the data you want to
-           construct the Centroids with to the constructor instead.
-
-        The geometry, lat, and lon attributes are set from the GeoDataFrame.geometry attribute,
-        while the columns are copied as attributes to the Centroids object in the form of
-        numpy.ndarrays using pandas.Series.to_numpy. The Series dtype will thus be respected.
-
-        Columns named lat or lon are ignored, as they would overwrite the coordinates extracted
-        from the point features. If the geometry attribute bears an alias, it can be dropped by
-        setting the geometry_alias parameter.
+    def to_crs(self, crs):
+        """ Project the current centroids to the default CRS (epsg4326)
+        Modifies the object in place.
 
-        If the GDF includes a region_id column, but no on_land column, then on_land=True is
-        inferred for those centroids that have a set region_id.
+        Parameters
+        ----------
+        crs : str
+            coordinate reference system
 
-        Example
+        Returns
         -------
-        >>> gdf = geopandas.read_file('centroids.shp')
-        >>> gdf.region_id = gdf.region_id.astype(int)  # type coercion
-        >>> centroids = Centroids.from_geodataframe(gdf)
+        Centroids
+            Centroids in the new crs
+        """
+        return Centroids.from_geodataframe(self.gdf.to_crs(crs))
+
+    @classmethod
+    def from_geodataframe(cls, gdf):
+        """Initialize centroids from a geodataframe
```

Comment:
We decided that the class and not the user should handle the layout of the underlying geodataframe. So I think that this classmethod should not just take an arbitrary geodataframe and create a new object from it.

Reply:
Done in [f59278c](https://github.com/CLIMADA-project/climada_python/pull/787/commits/f59278cf542473ccbb398ff1168d1068dd0447fd)